### PR TITLE
CC-33982: Added direct synchronization and flaxibale chunk size configuration to improve P&S performance.

### PR DESCRIPTION
**Pull request by dmiseev**
## PR Description

- Epic: https://spryker.atlassian.net/browse/CC-33982
- Doc: https://spryker.atlassian.net/browse/CC-34060

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Review by AsonUnique — APPROVED** [review]

File: docs/dg/dev/backend-development/data-manipulation/queue/queue.md [hunk 1]
```diff
@@ -129,6 +129,25 @@ class SampleQueueMessageProcessorPlugin implements QueueMessageProcessorPluginIn
 ?>
 ```
 
+### Configuration for chunk size
```
Comment (andriitserkovnyi) [inline docs/dg/dev/backend-development/data-manipulation/queue/queue.md:132]: Is this section related to Queue message processor plugin?

File: docs/dg/dev/backend-development/data-manipulation/queue/queue.md [hunk 2]
```diff
@@ -129,6 +129,25 @@ class SampleQueueMessageProcessorPlugin implements QueueMessageProcessorPluginIn
...
+    'publish' => 2000,
+    'event' => 1000,
+    'sync.search.product' => 100
+];
+?>
+```
+
+This configuration takes precedence over the `getChunkSize()` method. If a chunk size is defined for a queue in `QUEUE_MESSAGE_CHUNK_SIZE_MAP`, that size is used. Otherwise, the chunk size falls back to the value returned by the `getChunkSize()` method of the `QueueMessageProcessorPluginInterface`.
+
 Register the plugin in `QueueDependencyProvider::getProcessorMessagePlugins()`:
```
Comment (andriitserkovnyi) [inline docs/dg/dev/backend-development/data-manipulation/queue/queue.md:151]: How is this related to this section?

File: docs/dg/dev/backend-development/data-manipulation/queue/queue.md (hunk 1 above)
Comment (dmiseev) [reply docs/dg/dev/backend-development/data-manipulation/queue/queue.md:132]: yes, it's a flexible configuration for the processor plugin that we enhanced in this ticket.

File: docs/dg/dev/backend-development/data-manipulation/queue/queue.md (hunk 2 above)
Comment (dmiseev) [reply docs/dg/dev/backend-development/data-manipulation/queue/queue.md:151]: Processor plugins have some gaps related to chunk size configuration, so the new configuration is more prioritized/preferable than the default one.
